src/command_modules/azure-cli-sf/azure/cli/command_modules/sf/custom.py

```diff
@@ -0,0 +1,1480 @@
+    else:
+        set_global_config_value("servicefabric", "security", "none")
+
+    if ca:
+        set_global_config_value("servicefabric", "ca_path", ca)
+
+    if no_verify:
+        set_global_config_value("servicefabric", "no_verify", "True")
+    else:
+        set_global_config_value("servicefabric", "no_verify", "False")
+
+    set_global_config_value("servicefabric", "endpoint", endpoint)
+
+
+def sf_get_verify_setting():
+    az_config.config_parser.read(CONFIG_PATH)
+    no_verify = az_config.get("servicefabric", "no_verify", fallback="False")
+    return no_verify == "True"
+
+
+def sf_get_ca_cert_info():
+    az_config.config_parser.read(CONFIG_PATH)
+    ca_cert = az_config.get("servicefabric", "ca_path", fallback=None)
+    return ca_cert
+
+
+def sf_get_connection_endpoint():
+    az_config.config_parser.read(CONFIG_PATH)
+    return az_config.get("servicefabric", "endpoint", fallback=None)
+
+
+def sf_get_cert_info():
+    az_config.config_parser.read(CONFIG_PATH)
+    security_type = str(az_config.get("servicefabric",
+                                      "security", fallback=""))
+    if security_type == "pem":
+        pem_path = az_config.get("servicefabric", "pem_path", fallback=None)
+        return pem_path
+    elif security_type == "cert":
+        cert_path = az_config.get("servicefabric", "cert_path", fallback=None)
+        key_path = az_config.get("servicefabric", "key_path", fallback=None)
+        return cert_path, key_path
+    elif security_type == "none":
+        return None
+    else:
+        raise CLIError("Cluster security type not set")
+
+
+class FileIter:  # pylint: disable=too-few-public-methods
+    def __init__(self, file, rel_file_path):
+        self.file = file
+        self.rel_file_path = rel_file_path
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        chunk = self.file.read(100000)
+        if chunk == b'':
+            raise StopIteration
+        else:
+            return chunk
+
+
+def sf_upload_app(path):
+    """
+    Copies a Service Fabric application package to the image store.
+
+
+    The cmdlet copies a Service Fabric application package to the image store.
+    After copying the application package, use the sf application provision
+    cmdlet to register the application type.
+
+    :param str path: The path to your local application package
+    """
+    abspath = os.path.abspath(path)
+    basename = os.path.basename(abspath)
+    endpoint = sf_get_connection_endpoint()
+    cert = sf_get_cert_info()
+    ca_cert = False
+    if cert is not None:
+        ca_cert = sf_get_ca_cert_info()
+    total_files_count = 0
+    current_files_count = 0
+    total_files_size = 0
+    for root, _, files in os.walk(abspath):
+        total_files_count += (len(files) + 1)
+        for file in files:
+            t = os.stat(os.path.join(root, file))
+            total_files_size += t.st_size
```

Comment:
Two things to be done regarding to the progress indicator:
1) use a `--show-progress`, `--version` has its own meaning, it is associated with elevating logging level.
2) print the progress information to `stderr` instead of `stdout`, as other Unix tools do.